src/v/coproc/partition_manager.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/ntp_callbacks.h"
+#include "cluster/partition.h"
+#include "coproc/partition.h"
+#include "storage/fwd.h"
+
+#include <seastar/core/gate.hh>
+
+#include <absl/container/flat_hash_map.h>
+
+namespace coproc {
+
+class partition_manager {
+public:
+    using ntp_table_container
+      = absl::flat_hash_map<model::ntp, ss::lw_shared_ptr<partition>>;
+    using manage_cb_t
+      = ss::noncopyable_function<void(ss::lw_shared_ptr<partition>)>;
+    using unmanage_cb_t = ss::noncopyable_function<void(model::partition_id)>;
+
+    explicit partition_manager(ss::sharded<storage::api>& storage) noexcept;
+
+    ss::future<> start() { return ss::now(); }
+    ss::future<> stop();
```

Comment:
do we need a separate `shutdown` method ?, now when stop is called `ss::sharded<partition_manager>` instances will be destroyed.

Reply:
I have now modified this class to have the same lifetime as `cluster::partition_manager`